e2e/gui/src/test/java/com/epam/pipeline/autotests/ToolsParametersTest.java

```diff
@@ -54,11 +91,83 @@ public void runToolThatHaveNoNginxEndpoint() {
                                 .save()
                                 .run(this)
                 )
-                .showLog(getRunId());
+                .showLog(getLastRunId());
         new RunsMenuAO()
-                .waitForInitializeNode(getRunId())
+                .waitForInitializeNode(getLastRunId())
                 .clickEndpoint()
                 .screenshot("test501screenshot")
                 .assertPageTitleIs("502 Bad Gateway");
     }
+
+    @Test
+    @TestCase(value = {"2234"})
+    public void customCapabilitiesImplementation() {
+        logoutIfNeeded();
+        loginAs(admin);
+        navigationMenu()
+                .settings()
+                .switchToPreferences()
+                .clearAndSetJsonToPreference(launchCapabilities,
+                        readResourceFully(CUSTOM_CAPABILITIES_1_JSON), true)
+                .saveIfNeeded();
+        logout();
+        loginAs(user);
+        tools()
+                .perform(registry, group, tool, ToolTab::runWithCustomSettings)
+                .expandTab(EXEC_ENVIRONMENT)
+                .selectValue(RUN_CAPABILITIES, custCapability1)
+                .click(DISK)
+                .selectValue(RUN_CAPABILITIES, custCapability2)
+                .checkTooltipText(custCapability1, "Custom test capability 1")
+                .checkTooltipText(custCapability2, "Custom test capability 2")
+                .launch(this)
+                .showLog(getLastRunId())
+                .expandTab(PARAMETERS)
+                .ensure(configurationParameter(format("CP_CAP_CUSTOM_%s", custCapability1), "true"), exist)
+                .ensure(configurationParameter(format("CP_CAP_CUSTOM_%s", custCapability2), "true"), exist)
+                .waitForSshLink()
+                .click(taskWithName("Console"))
+                .waitForLog("start.sh")
+                .ensure(log(), matchText(format("Running '%s' commands:", custCapability1)))
```

Comment:
"Running '%s' commands:" can be constant

Reply:
Done